### PR TITLE
[packager] bugfix #3997: react-native bundle not working on windows 10

### PR DESCRIPTION
**Pull request by bramstieperaere**
fixes https://github.com/facebook/react-native/issues/3997

the root cause is in

Mon, 09 Nov 2015 13:22:47 GMT ReactNativePackager:SocketServer uncaught error Error: listen EACCES C:\Users\donald\AppData\Local\Temp\react-packager-9248a9803ac72b509b389b456696850d

This means that the socket server cannot create the socket.

cfr https://gist.github.com/domenic/2790533 the socket name is not a valid windows socket name.

### CODE REVIEW

File: packager/react-packager/src/SocketInterface/index.js [hunk 1]
```diff
@@ -35,10 +35,15 @@ const SocketInterface = {
         }
       });
 
-      const sockPath = path.join(
+      let sockPath = path.join(
         tmpdir,
         'react-packager-' + hash.digest('hex')
       );
+      if (process.platform==='win32'){
```
Comment (mkonicek) [inline packager/react-packager/src/SocketInterface/index.js]: nit: spaces around ===

File: packager/react-packager/src/SocketInterface/index.js [hunk 2]
```diff
@@ -35,10 +35,15 @@ const SocketInterface = {
         }
       });
 
-      const sockPath = path.join(
+      let sockPath = path.join(
```
Comment (mkonicek) [inline packager/react-packager/src/SocketInterface/index.js]: Why `let`?

File: packager/react-packager/src/SocketInterface/index.js [hunk 3]
```diff
@@ -35,10 +35,15 @@ const SocketInterface = {
...
         tmpdir,
         'react-packager-' + hash.digest('hex')
       );
+      if (process.platform==='win32'){
+        sockPath = sockPath.replace(/^\//, '')
+        sockPath = sockPath.replace(/\//g, '-')
```
Comment (mkonicek) [inline packager/react-packager/src/SocketInterface/index.js]: Are forward slashes not allowed in paths on Windows? Consider adding a short comment explaining these two lines.

File: packager/react-packager/src/SocketInterface/index.js [hunk 4]
```diff
@@ -35,10 +35,15 @@ const SocketInterface = {
...
+        sockPath = '\\\\.\\pipe\\' + sockPath
```
Comment (mkonicek) [inline packager/react-packager/src/SocketInterface/index.js]: Is `\\.\pipe\` some special path on Windows? Can you add a comment explaining this?

**Comment by mkonicek** [comment]
lgtm but don't have enough context on this code, I'll let @davidaurelio review.

**Comment by mkonicek** [comment]
Thanks a lot for fixing this!

File: packager/react-packager/src/SocketInterface/index.js (hunk 2 above)
Comment (bramstieperaere) [inline packager/react-packager/src/SocketInterface/index.js]: cannot use const, the variable is assigned another value in lines 43-44. I thought let would be safest (smallest scope)

File: packager/react-packager/src/SocketInterface/index.js (hunk 4 above)
Comment (bramstieperaere) [inline packager/react-packager/src/SocketInterface/index.js]: I copied this code from https://gist.github.com/domenic/2790533

as far as I understand, it creates a named pipe by converting the sockPath variable in a valid pipe name by replacing illegal characters for a named pipe (line 43-44), and then prefixes with the path for a named pipe on local machine. (line 45)

File: local-cli/bundle/saveBundleAndMap.js [hunk 5]
```diff
@@ -84,13 +85,14 @@ function copyAll(filesToCopy) {
...
+
+  mkdirp(destDir, err => {
     if (err) {
       return callback(err);
     }
     fs.createReadStream(src)
-      .pipe(fs.createWriteStream(dest))
-      .on('finish', callback);
+        .pipe(fs.createWriteStream(dest))
+        .on('finish', callback);
```
Comment (davidaurelio) [inline local-cli/bundle/saveBundleAndMap.js]: nit: can you remove the additional whitespace, please?

**Comment by davidaurelio** [comment]
@bramstieperaere thank you for flagging + fixing this! I’d like to get our travis tests back to green before I merge this. I will ping you and ask you to rebase on top of master when we are there.

**Comment by mkonicek** [comment]
Thanks @davidaurelio :+1:

**Comment by davidaurelio** [comment]
@bramstieperaere travis is green again. Would you please rebase on or merge master and address the nitpicks we have? Thanks a lot!

File: packager/react-packager/src/SocketInterface/index.js (hunk 4 above)
Comment (davidaurelio) [inline packager/react-packager/src/SocketInterface/index.js]: Would you add a comment where the code came from, please?

**Comment by bramstieperaere** [comment]
@davidaurelio 

Rebased, added the requested changes from code review.

**Comment by davidaurelio** [comment]
@facebook-github-bot import

**Comment by davidaurelio** [comment]
Something is fishy about the amount of changes in this diff. Checking it internally.

**Pull request closed without merging**